### ISSUE TITLE
Zero coverage on svelte components

### ISSUE THREAD

**Issue by MoritzHorch** [vitest-dev/vitest#1828]
### Describe the bug

Hello, I followed your example for the setup for Svelte as seen here: https://github.com/vitest-dev/vitest/tree/main/examples/svelte 

For automated testing and pipelines, I wanted to setup a coverage threshold. As documented, I added `c8` to my dependencies. When running `vitest run --coverage` and having the following configuration, I get zero coverage on my components / files.

`vitest.config.ts`:
```typescript
import { defineConfig } from 'vite';
import { svelte } from '@sveltejs/vite-plugin-svelte';

export default defineConfig({
  plugins: [svelte({ hot: !process.env.VITEST })],
  test: {
    globals: true,
    environment: 'jsdom',
    coverage: {
      all: true,
      include: ['components/**'],
      reporter: ['text', 'lcov'],
    },
  },
});
```

console output:
```bash
 RUN  v0.21.1 /home/projects/vitest-dev-vitest-j7shma

 ✓ test/basic.test.ts (1)
 ✓ test/hello.test.ts (2)

Test Files  2 passed (2)
     Tests  3 passed (3)
  Start at  20:13:24
  Duration  4.61s (setup 0ms, collect 530ms, tests 98ms)

--------------|---------|----------|---------|---------|-------------------
File          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
--------------|---------|----------|---------|---------|-------------------
All files     |       0 |        0 |       0 |       0 |                   
 Hello.svelte |       0 |        0 |       0 |       0 | 1-7               
--------------|---------|----------|---------|---------|-------------------
close timed out after 1000ms
```

I expect full coverage as everything is tested. This should not be the behavior, right? I hardly changed your given example.

I have seen that there are issues for not exact coverage, but personally I am missing all of it. On the Discord server I did not recieve any further help / solutions.

### Reproduction

Slightly adjusted your example: [https://stackblitz.com/edit/vitest-dev-vitest-j7shma](https://stackblitz.com/edit/vitest-dev-vitest-j7shma)

### System Info

```shell
❯ npx envinfo --system --npmPackages '{vitest,vite,@vitejs/*}' --binaries --browsers
success Install finished in 1.366s


  System:
    OS: Linux 5.0 undefined
    CPU: (4) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.10 - /bin/yarn
    npm: 7.17.0 - /bin/npm
  npmPackages:
    vitest: latest => 0.21.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by AriPerkkio** [comment]
@MoritzHorch does this happen on your local machine as well? Stackblitz does not implement all of the `v8`'s APIs, such as `takeCoverage`. `c8` won't work there.

By downloading and running locally the repro project I'm seeing following:
```
--------------|---------|----------|---------|---------|-------------------
File          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
--------------|---------|----------|---------|---------|-------------------
All files     |     100 |        0 |     100 |     100 |                   
 Hello.svelte |     100 |        0 |     100 |     100 | 6                 
--------------|---------|----------|---------|---------|-------------------
```

**Comment by MoritzHorch** [comment]
@AriPerkkio Indeed, locally it just worked. In my project it still didn't work until, after some trial and error, I found the problem.

I used a lot of "Create React App" for work and therefore took absolute imports from the root project folder for granted. Such things as `import App from 'src/App.svelte`. I now learned that this is not the default behaviour with Svelte + Vite / Vitest. `c8` probably wasn't able to resolve / follow this path and couldn't collect any coverage.

Therefore **the solution** 🎉 was to either use relative imports like `import App from '../src/App.svelte` or use [vite-tsconfig-paths](https://github.com/aleclarson/vite-tsconfig-paths). I did configured it as described and it simply worked. If I understood this plugin correctly, it will additionally do another lookup starting from the configured `baseUrl` in the `tsconfig.json` (`"baseUrl": "."` in my case).

Guess my given example simply did not work due to the reason you described, but my real problem was the missing configuration / wrong use which lead to a lot of confusion. 

Thanks for helping and thanks to "hylo" on the Svelte Discord Server as well!
